src/java/org/apache/cassandra/service/accord/AccordConfigurationService.java

```diff
@@ -67,23 +66,24 @@
 @Simulate(with=MONITORS)
 public class AccordConfigurationService extends AbstractConfigurationService<AccordConfigurationService.EpochState, AccordConfigurationService.EpochHistory> implements ChangeListener, AccordEndpointMapper, AccordSyncPropagator.Listener, Shutdownable
 {
-    private static final Logger logger = LoggerFactory.getLogger(AccordConfigurationService.class);
-
     private final AccordSyncPropagator syncPropagator;
     private final DiskStateManager diskStateManager;
 
+    @GuardedBy("this")
     private EpochDiskState diskState = EpochDiskState.EMPTY;
 
     private enum State { INITIALIZED, LOADING, STARTED, SHUTDOWN }
 
+    @GuardedBy("this")
     private State state = State.INITIALIZED;
     private volatile EndpointMapping mapping = EndpointMapping.EMPTY;
 
     public enum SyncStatus { NOT_STARTED, NOTIFYING, COMPLETED }
 
     static class EpochState extends AbstractConfigurationService.AbstractEpochState
     {
-        SyncStatus syncStatus = SyncStatus.NOT_STARTED;
+        @GuardedBy("AccordConfigurationService.this")
```

Comment:
this is checked then we update disk state... so sharing the lock keeps reasoning about correctness simpler